aea/helpers/async_friendly_queue.py

```diff
@@ -41,7 +41,9 @@ def put(self, item: Any, *args, **kwargs) -> None:
         super().put(item, *args, **kwargs)
         if self._non_empty_waiters:
             waiter = self._non_empty_waiters.popleft()
-            waiter._loop.call_soon_threadsafe(waiter.set_result, True)
+            waiter._loop.call_soon_threadsafe(  # pylint: disable=protected-access
```

Comment:
If we didn't have to be compatible with Python 3.6, we could have used `get_loop()`:
https://docs.python.org/3/library/asyncio-future.html#asyncio.Future.get_loop

Reply:
But we do have to be compatible :)